engines/agi/op_cmd.cpp

```diff
@@ -32,8 +32,29 @@
 #include "agi/words.h"
 
 #include "common/random.h"
+#include "common/system.h"
 #include "common/textconsole.h"
 
+namespace {
+// Creates a unique log file name each time this function is called.
+// We never want to override an existing log file, so here we create one
+// based on the current game and system time.
+//
+// For example: agi.kq.20221013214511.log
+Common::String generateLogFileName(Agi::AgiGame *state, Agi::AgiEngine *vm) {
+	TimeDate date;
+	vm->_system->getTimeAndDate(date, true);
+	return Common::String::format("agi.%s.%d%02d%02d%02d%02d%02d.log",
```

Comment:
Please put it in the "dumps/" directory. The users will need to create that directory manually if the want to see the logs. That is a common practice in ScummVM.